src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -384,13 +390,43 @@ private static Vector2 PatchCenter(Vector2 pos)
             pos.Y + Constants.PATCH_NODE_RECT_LENGTH * 0.5f);
     }
 
-    private Line2D CreateConnectionLine(Vector2[] points, Color connectionColor)
+    private static Vector2 ClosestPatchSideCenter(Vector2 patchPos, Vector2 pointPos, bool isVertical)
+    {
+        float halfLength = Constants.PATCH_NODE_RECT_LENGTH * 0.5f;
+        float posX = patchPos.X;
+        float posY = patchPos.Y;
+        List<Vector2> sideCenterPoints = new();
+        if (!isVertical)
+        {
+            // Right
+            sideCenterPoints.Add(new Vector2(posX + 2 * halfLength, posY + halfLength));
+
+            // Left
+            sideCenterPoints.Add(new Vector2(posX, posY + halfLength));
+        }
+        else
+        {
+            // Top
+            sideCenterPoints.Add(new Vector2(posX + halfLength, posY));
+
+            // Bottom
+            sideCenterPoints.Add(new Vector2(posX + halfLength, posY + 2 * halfLength));
+        }
+
+        return sideCenterPoints.OrderBy(point => point.DistanceSquaredTo(pointPos)).First();
+    }
+
+    private Line2D CreateConnectionLine(Vector2[] points, Color connectionColor, int zIndex = 1,
+        float width = Constants.PATCH_REGION_CONNECTION_LINE_WIDTH)
     {
         var link = new Line2D
         {
             DefaultColor = connectionColor,
             Points = points,
-            Width = Constants.PATCH_REGION_CONNECTION_LINE_WIDTH,
+            Width = width,
+            BeginCapMode = Line2D.LineCapMode.Round,
+            EndCapMode = Line2D.LineCapMode.Round,
+            ZIndex = zIndex,
```

Comment:
ZIndex can mess with other layers than purely just this. For example even the pause menu might not work correctly. So I'd really prefer an approach where ZIndex is not used. Reordering the children in the line container can be used to set the draw order. If just a couple of ZIndex values are needed, they could be made as separate containers for the lines. Or what is it needed for? I can probably come up with some better suggestions if I know that.